### PR TITLE
integration of redux with dark mode

### PR DESCRIPTION
**Pull request by betomoedano**
Integration of Redux ⚛️

- create store
- create reducer for dark mode
- create custom hooks
- basic dark mode functionality
- custom header component

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,17 +1,30 @@
 import "./App.css";
+import Header from "./components/Header";
+import { useAppSelector } from "../src/app/hooks"
```
Comment (cesswhite) [inline src/App.tsx]: import for Header component is not necessary

File: src/App.tsx [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
 function App() {
 
+  const isDarkMode = useAppSelector(state => state.darkMode.isDarkMode);
+
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <h1 className="text-7xl">
+    <div style={{
```
Comment (cesswhite) [inline src/App.tsx:12]: Can you make this a component and call it here?

File: src/App.tsx [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
+      backgroundColor: isDarkMode ? '#181818' : '#fff',
+      color: isDarkMode ? '#fff' : '#181818',
+      width: '100vw',
+      height: '100vh',
```
Comment (cesswhite) [inline src/App.tsx]: Please, use tailwind

File: src/app/hooks.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
+import { RootState, AppDispatch } from "./store";
+
+export const useAppDispatch = () => useDispatch<AppDispatch>();
+export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
```
Comment (cesswhite) [inline src/app/hooks.ts:5]: add line break

File: src/app/store.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+import darkModeReducer from "../features/darkMode/dark-mode-slice";
+
+export const store = configureStore({
+    reducer: {
+        darkMode: darkModeReducer,
+    },
+});
+
+export type AppDispatch = typeof store.dispatch;
+export type RootState = ReturnType<typeof store.getState>;
```
Comment (cesswhite) [inline src/app/store.ts:11]: add line break

File: src/components/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            <nav className="fixed w-full top-0 z-10 bg-white backdrop-filter backdrop-blur-lg bg-opacity-30 border-b border-gray-200">
+                <div className="max-w-5xl mx-auto px-4">
+                    <div  className="flex items-center justify-between h-16">
+                        <span className="text-2xl text-gray-900 font-semibold"></span>
+                        <div className="flex space-x-4 text-gray-900">
+                            {/* <a href="#"> Home </a>
+                            <a href="#"> Users </a>
+                            <a href="#"> Projects </a>
+                            <a href="#"> Contact </a> */}
+                            <button className="" onClick={() => dispatch(toggleDarkMode())}>
```
Comment (cesswhite) [inline src/components/Header.tsx]: can you create a component of this logic and call it here?

File: src/features/darkMode/dark-mode-slice.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        },
+        
+        toggleDarkMode: (state) => {
+            state.isDarkMode = !state.isDarkMode;
+        }
+    }
+});
+
+export const { setDarkMode, toggleDarkMode } = darkModeSlice.actions;
+export default darkModeSlice.reducer;
```
Comment (cesswhite) [inline src/features/darkMode/dark-mode-slice.ts]: end line

**Review by cesswhite — COMMENTED** [review]
Please, resolve the comments, thank you

File: src/App.tsx (hunk 1 above)
Comment (betomoedano) [reply src/App.tsx]: I tried to removed but it does not work

**Pull request merged**